era5_monthly_averages_single_level_1979_2024/process_downloads.jl

```diff
@@ -0,0 +1,319 @@
+                "longitude",
+                "date",
+                "msdwlwrf",
+                "msdwswrf",
+                "msnlwrf",
+                "msnswrf",
+                "expver",
+                "number",
+            ]
+        )
+            attrib = copy(var.attrib)
+            for (key, value) in attrib
+                if key in ignored_attribs ||
+                   occursin("GRIB", key) ||
+                   attrib[key] == "unknown"
+                    delete!(attrib, key)
+                end
+            end
+            # the _FillValue attribute is automatically added by NCDatasets
+            # store everything as Float32 to save space, and max compression (lossless)
+            defVar(
+                output_ds,
+                varname,
+                Float32.(reverse(var[:, :, :], dims = 2)),
+                (dimnames(var)[1:2]..., "time");
+                attrib = attrib,
+                deflatelevel = 9,
+            )
+        end
+    end
+    defVar(
+        output_ds,
+        "expver",
+        input_ds["expver"][:],
+        ("time",);
+        attrib = input_ds["expver"].attrib,
+    )
+    defVar(
+        output_ds,
+        "msuwlwrf",
+        Float32.(
+            reverse(
+                input_ds["msdwlwrf"][:, :, :] .- input_ds["msnlwrf"][:, :, :],
+                dims = 2,
+            )
+        ),
+        ("longitude", "latitude", "time"),
+        attrib = (
+            "units" => "W m**-2",
+            "long_name" => "Mean surface upward long-wave radiation flux",
+        ),
+        deflatelevel = 9,
+    )
+    defVar(
+        output_ds,
+        "msuwswrf",
+        Float32.(
+            reverse(
+                input_ds["msdwswrf"][:, :, :] .- input_ds["msnswrf"][:, :, :],
+                dims = 2,
+            )
+        ),
+        ("longitude", "latitude", "time"),
+        attrib = (
+            "units" => "W m**-2",
+            "long_name" => "Mean surface upward short-wave radiation flux",
+        ),
+        deflatelevel = 9,
+    )
+
+    defVar(
+        output_ds,
+        "latitude",
+        reverse(input_ds["latitude"][:]),
+        ("latitude",);
+        attrib = delete!(copy(input_ds["latitude"].attrib), "stored_direction"),
+    )
+
+    defVar(
+        output_ds,
+        "longitude",
+        input_ds["longitude"][:],
+        ("longitude",);
+        attrib = input_ds["longitude"].attrib,
+    )
+
+    new_times = map(input_ds["date"][:]) do t
+        d = DateTime(t)
+        d + (Day(15) - Day(d))
+    end
+
+    # check that there are no duplicates and that it is sorted
+    @assert issorted(new_times)
+    for i = 2:length(new_times)
+        @assert new_times[i] != new_times[i-1]
+    end
+    new_times_attribs = ["standard_name" => "time", "long_name" => "Time"]
+
+    defVar(output_ds, "time", new_times, ("time",); attrib = new_times_attribs)
+    close(output_ds)
+end
+
+convert_to_f32(x) = Float32(x)
+convert_to_f32(x::Missing) = x
```

Comment:
Out of curiosity, when was there missing data?

Reply:
maybe runoff over ocean?

Reply:
Downloading Monthly averages by hour of day results in some weirdness with the time dimension, where there are duplicate entries. Some variables are defined on one of the duplicates and some on the other. I had improperly dealt with this (assumed all were on one of the duplictes), which resulted in missing data. This is fixed now, and there is no missing data